homeassistant/components/noaa_tides/sensor.py

```diff
@@ -49,41 +60,138 @@
 )
 
 
-def setup_platform(
+async def async_setup_platform(
     hass: HomeAssistant,
     config: ConfigType,
     add_entities: AddEntitiesCallback,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the NOAA Tides and Currents sensor."""
+
+    _LOGGER.debug("Importing from configuration.yaml: %s", config)
+
     station_id = config[CONF_STATION_ID]
-    name = config.get(CONF_NAME)
-    timezone = config.get(CONF_TIME_ZONE)
+    timezone = config.get(CONF_TIME_ZONE, DEFAULT_TIMEZONE)
+    unit_system = config.get(CONF_UNIT_SYSTEM, get_default_unit_system(hass))
 
-    if CONF_UNIT_SYSTEM in config:
-        unit_system = config[CONF_UNIT_SYSTEM]
-    elif hass.config.units is METRIC_SYSTEM:
-        unit_system = UNIT_SYSTEMS[1]
-    else:
-        unit_system = UNIT_SYSTEMS[0]
+    user_data = {
+        CONF_STATION_ID: station_id,
+        CONF_TIME_ZONE: timezone,
+        CONF_UNIT_SYSTEM: unit_system,
+    }
 
-    try:
-        station = coops.Station(station_id, unit_system)
-    except KeyError:
-        _LOGGER.error("NOAA Tides Sensor station_id %s does not exist", station_id)
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_IMPORT},
+        data=user_data,
+    )
+
+    if result["type"] is FlowResultType.CREATE_ENTRY or (
+        result["type"] is FlowResultType.ABORT
+        and result["reason"] == "already_configured"
+    ):
+        async_create_issue(
+            hass,
+            HOMEASSISTANT_DOMAIN,
+            f"deprecated_yaml_{DOMAIN}",
+            breaks_in_ha_version=DEPRECATE_YAML_IN_HA_VERSION,
+            is_fixable=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key="deprecated_yaml",
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": NAME,
+            },
+        )
         return
+
+    if (
+        result["type"] is FlowResultType.FORM
+        and result["errors"] is not None
+        and result["errors"]["base"] is not None
+    ):
+        _LOGGER.error(
```

Comment:
Ideally we just abort the flow instead of returning it in an error state, I think there's more logic to that

Reply:
I'm not sure what to do here.

Should I just remove this entire if block and let it fallback to the unknown error?

If it is better to abort the flow, how do I do that? Just "return None"? Shouldn't we shomehow inform the user that the import failed?

I'm trying to handle the "station_not_found" and "cannot_connect" errors on import. I think I stole this idea from another integration but I can't remember which one - and I can't find it anymore.